rfd/0241-scoped-access-lists.md

```diff
@@ -0,0 +1,931 @@
+In this case if 20k users were members of any number of lists across 20 scopes,
+there would be 400k materialized assignments.
+The downside here would be that the aggregated scoped roles assignments could
+become large, and the source of the materialized assignment would be more
+difficult to reason about.
+The current plan is not to aggregate scoped role assignments.
+
+Earlier in the design for scoped access lists there was discussion of writing
+these materialized assignments out to the cluster's backend database.
+For the time being, we are opting not to do this and instead keep the
+materialized assignments in the scoped access cache only.
+
+Some reasons not to store materialized assignments in the backend are:
+
+1. The source of truth is the access list and member resources, storing the
+   materialized assignments would store redundant data, with redundancy there
+   is potential for drift or conflicts that need to be resolved.
+2. Multiple auth servers would be trying to write the same materialized
+   assignments to the backend, they may disagree or conflict, especially if
+   auth servers are running different versions.
+3. Dangling assignments would need to be detected and cleaned, while avoiding
+   deletion of new assignments created by a different auth server.
+4. Materialized assignments need to be computed and cached in memory on each
+   auth server anyway to keep the backend up to date.
+5. Keeping them out of the backend will reduce read and write load on the backend.
+6. Loading many resources from the backend can be very slow on startup, e.g. 5
+   minutes to read 2 million resources from dynamo. Recomputing is likely to be
+   faster.
+7. Everything that needs to read scoped role assignments already needs to use
+   the custom scoped access cache for scoped traversal, there are not going to
+   be access paths that read directly from the backend and bypass the cache.
+
+### Notable differences from traditional access lists
+
+Traditional Teleport access lists are capable of some fairly advanced features,
+including modifying user traits and delegating responsibility for management of
+the list to owners who otherwise do not have access list modification/creation
+permissions.
+
+We will be initially omitting all traditional access list features not strictly
+necessary for the goal of granting common sets of scoped privileges to groups
+of users.
+Additionally, some access list features are categorically incompatible with the
+nature of scopes and scoped role assignment, and some features will need to
+function differently under the hood.
+Some notable differences include:
+
+- There will be no support for access list owners.
+  Scoped access lists will rely on standard scoping rules to determine who can
+  manage them.
+  A concept of owners may be added in the future, but would likely not add
+  significant additional value when scoped admins can already manage access lists
+  and role assignments directly.
```

Comment:
The access list ownership is quite critial part of the access list concpet and allows to solve many day customers probleme like mangina JIT flow granting